cores/esp8266/sntp-lwip2.cpp

```diff
@@ -99,368 +99,26 @@ int settimeofday(const struct timeval* tv, const struct timezone* tz)
-	    m_day -= DAYSPERWEEK;
-
-	  days += m_day;
-	}
-
-      /* store the change-over time in GMT form by adding offset */
-      sntp__tzrule[i].change = days * SECSPERDAY + sntp__tzrule[i].s + sntp__tzrule[i].offset;
-    }
-
-  __tznorth = (sntp__tzrule[0].change < sntp__tzrule[1].change);
-
-  return 1;
-}
-
-char* sntp_asctime_r(struct tm *tim_p ,char *result)
-{
-  static const char day_name[7][4] = {
-	"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"
-  };
-  static const char mon_name[12][4] = {
-	"Jan", "Feb", "Mar", "Apr", "May", "Jun",
-	"Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
-  };
-  os_sprintf (result, "%s %s %02d %02d:%02d:%02d %02d\n",
-	   day_name[tim_p->tm_wday],
-	   mon_name[tim_p->tm_mon],
-	   tim_p->tm_mday, tim_p->tm_hour, tim_p->tm_min,
-	   tim_p->tm_sec, 1900 + tim_p->tm_year);
-  return result;
-}
-
-char* sntp_asctime(struct tm *tim_p)
-{
-    return sntp_asctime_r (tim_p, reult);
-}
-
-uint32 ICACHE_RAM_ATTR sntp_get_current_timestamp(void)
-{
-    return realtime_stamp;
-}
-
-char* sntp_get_real_time(time_t t)
-{
-    return sntp_asctime(sntp_localtime (&t));
-}
-
-/* Returns the set timezone in seconds. If the timezone was set as seconds, the fractional part is floored. */
-sint32 sntp_get_timezone_in_seconds(void)
-{
-    return time_zone;
-}
-
-/* Returns the set timezone in hours. If the timezone was set as seconds, the fractional part is floored. */
-sint8 sntp_get_timezone(void)
-{
-    return (sint8)(time_zone / (60 * 60));
-}
-
-/* Sets the timezone in hours. Internally, the timezone is converted to seconds. */
-bool sntp_set_timezone_in_seconds(sint32 timezone)
-{
-    if(timezone >= (-11 * (60 * 60)) || timezone <= (13 * (60 * 60))) {
-        time_zone = timezone;
-        return true;
-    }
-    return false;
-}
-
-/* Sets the timezone in hours. Internally, the timezone is converted to seconds. */
-bool sntp_set_timezone(sint8 timezone)
-{
-    return sntp_set_timezone_in_seconds((sint32)timezone * 60 * 60);
-}
-
-		   
-void sntp_set_daylight(int daylight)
-{
-    dst = daylight;
-}
-
 void ICACHE_RAM_ATTR sntp_time_inc (void)
 {
-    realtime_stamp++;
-}
-
-static void sntp_set_system_time (uint32_t t)
-{
-    realtime_stamp = t + time_zone + dst;
-    os_timer_disarm(&sntp_timer);
-    os_timer_setfn(&sntp_timer, (os_timer_func_t *)sntp_time_inc, NULL);
-    os_timer_arm(&sntp_timer, 1000, 1);
+    sntp_real_timestamp++;
 }
 
 int settimeofday(const struct timeval* tv, const struct timezone* tz)
 {
-    if (tz) /*before*/
-    {
-        sntp_set_timezone_in_seconds(tz->tz_minuteswest * 60);
-        // apparently tz->tz_dsttime is a bitfield and should not be further used (cf man)
-        sntp_set_daylight(0);
-    }
+    (void)tz; // obsolete (cf. man settimeofday)
     if (tv) /*after*/
     {
         // reset time subsystem
         tune_timeshift64(tv->tv_sec * 1000000ULL + tv->tv_usec);
 
-        sntp_set_system_time(tv->tv_sec);
+		sntp_real_timestamp = tv->tv_sec;
```

Comment:
This seems to rely on a OS timer to handle second ticks.  Is this not going to drift like mad?  If so, does it matter (i.e. are you just looking for a monotonically increasing int somewhere)?

Reply:
It is like that since the beginning (patched into lwip1 by espressif).
I was thinking this has the same accuracy than anything else handled by the OS or cpu even the cycle counter because everything is driven by the local crystal. Do you think we can have a better accuracy with another local source ?
SNTP readjusts it every hour anyway (by default).

Reply:
> Timer APIs can be found in/ESP8266_NONOS_SDK/include/osapi.h. Please note that os_timer APIs listed below are software timers executed in tasks, thus timer callbacks may not be precisely executed at the right time; it depends on priority. If you need a precise timer, please use a hardware timer which can be executed in hardware interrupt.

From the API reference at https://www.espressif.com/sites/default/files/documentation/2c-esp8266_non_os_sdk_api_reference_en.pdf

So it's not IRQ driven, but from the description I can't say if it's based on a HW timer (ie. you'll get jitter but not miss anything).  Since you say it's reset every 3600 seconds, I guess it's not a problem.

Reply:
I think `"may not be precisely executed at the right time"` would indeed mean that it may introduce jitter but the frequency average would stay still (or have the same drift as with HW timers).